web_transport/sdk/impl/web_transport_stream_impl.cc

```diff
@@ -120,22 +120,85 @@ size_t WebTransportStreamImpl::Read(uint8_t* data, size_t length) {
 }
 
 size_t WebTransportStreamImpl::ReadableBytes() const {
-  return stream_->ReadableBytes();
+  if (io_runner_->BelongsToCurrentThread()) {
+    return stream_->ReadableBytes();
+  }
+  size_t result;
+  base::WaitableEvent done(base::WaitableEvent::ResetPolicy::AUTOMATIC,
+                           base::WaitableEvent::InitialState::NOT_SIGNALED);
+  io_runner_->PostTask(
+      FROM_HERE,
+      base::BindOnce(
+          [](WebTransportStreamImpl const* stream, size_t& result,
+             base::WaitableEvent* event) {
+            result = stream->stream_->ReadableBytes();
+            event->Signal();
+          },
+          base::Unretained(this), std::ref(result), base::Unretained(&done)));
```

Comment:
Overall I have a question for these posted tasks... Is there a reason to use base::Unretained instead of WeakPtr?

Reply:
Yes, it's better to be WeakPtr. Many QUIC objects are not disposed actually. I've filed some issues in both QUIC SDK and server side for cleaning up. Raw pointers will be replace with `WeakPtr`s when necessary during cleaning up.